src/v/cluster/metrics_reporter.cc

```diff
@@ -187,9 +187,18 @@ ss::future<> metrics_reporter::start() {
 ss::future<> metrics_reporter::stop() {
     vlog(clusterlog.info, "Stopping Metrics Reporter...");
     _tick_timer.cancel();
+    _cluster_info_initialized_cvar.broken();
     co_await _gate.close();
 }
 
+ss::future<>
+metrics_reporter::wait_cluster_info_initialized(ss::abort_source& as) {
+    while (!as.abort_requested() && !_cluster_info.is_initialized()) {
+        co_await _cluster_info_initialized_cvar.wait(
+          5s, [this] { return _cluster_info.is_initialized(); });
```

Comment:
is the intention here to recheck after 5 seconds, or to actually throw a timeout exception after 5 seconds?